File: app/app/src/main/java/com/example/hakonsreader/api/model/ImgurAlbum.java
package com.example.hakonsreader.api.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Model for an Imgur album
 */
public class ImgurAlbum {

    @SerializedName("data")
    private Data data;


    private static class Data {
        @SerializedName("images")
        private List<Image> images;
    }

    /**
     * @return The list of images in this album (this size can be 1)
     */
    public List<Image> getImages() {
        return data.images;
    }
}

File: app/app/src/main/java/com/example/hakonsreader/api/model/ImgurAlbum.kt
package com.example.hakonsreader.api.model

import com.google.gson.annotations.SerializedName

/**
 * Model for an Imgur album
 */
class ImgurAlbum {
    @SerializedName("data")
    private val data: Data? = null

    private class Data {
        @SerializedName("images")
        val images: List<Image>? = null
    }

    /**
     * @return The list of images in this album (this size can be 1)
     */
    val images: List<Image>?
        get() = data?.images
}